Mathlib/Data/ByteArray.lean

```diff
@@ -80,6 +90,7 @@ def ByteArray.toSlice (arr : ByteArray) : ByteSlice := ⟨arr, 0, arr.size⟩
 
 /-- Convert a string of assumed-ASCII characters into a byte array.
 (If any characters are non-ASCII they will be reduced modulo 256.) -/
+@[deprecated (since := "2024-08-19")]
```

Comment:
@digama0, is there a replacement for this elsewhere that you know of?

Reply:
No. AFAIK the only reason it's being deprecated is because the semantics are somewhat weird in the non-ascii case.

Reply:
Right, the semantics are pretty strange indeed; but if a `String.toAscii? : Option ByteArray` existed, it would probably be sane to mention in this message.